src/energyplus/ForwardTranslator/ForwardTranslateElectricLoadCenterDistribution.cpp

```diff
@@ -63,26 +63,31 @@ namespace energyplus {
 
     IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::ElectricLoadCenter_Distribution, modelObject);
 
+    /// Generator Fields
     IdfObject generatorsIdf(openstudio::IddObjectType::ElectricLoadCenter_Generators);
-    generatorsIdf.setName(idfObject.name().get() + " Generators");
-    m_idfObjects.push_back(generatorsIdf);
+    if (!modelObject.generators().empty()) {
+      m_idfObjects.push_back(generatorsIdf);
```

Comment:
Probably move the line 67 inside the if block?

Reply:
Along with the other generator stuff?